packages/ai-jsx/src/batteries/constrained-output.tsx

```diff
@@ -78,9 +79,18 @@ export type TypedObjectCompletionWithRetry = TypedObjectCompletion & { retries?:
  *    Intermediate results that are valid, are also yielded.
  */
 export async function* JsonChatCompletion(
-  props: Omit<TypedObjectCompletionWithRetry, 'typeName' | 'parser' | 'partialResultCleaner'>,
+  { schema, ...props }: Omit<TypedObjectCompletionWithRetry, 'typeName' | 'parser' | 'partialResultCleaner'>,
   { render }: AI.ComponentContext
 ) {
+  if (schema) {
+    try {
+      return yield* render(<JsonChatCompletionFunctionCall schema={schema} {...props} />);
```

Comment:
feels like there's probably a formulation with `<ErrorBoundary>` here that's a little nicer but I think it amounts to the same thing

Reply:
You're right, but since I haven't used `ErrorBoundary` before and I'm not actively playing with this right now (so I won't find the issues), I'd rather stick to this for now.